content/master/guides/extensions-release-process.md

```diff
@@ -0,0 +1,301 @@
+2. Tag the desired commit on release branch with a valid semver release tag.
+For example, `v0.1.0`. By default, this is the inferred reference pushed to the registry.
+3. Manually run the workflow in the GitHub UI, targeting the release branch from step 1.
+
+See [branching conventions](#branching-conventions) for more details on tagging
+practices and optionally overriding the inferred git tag version.
+
+## Common Configuration
+
+While the reusable workflows referenced in the quickstart guides are for
+convenience, users may choose to write their own custom GitHub Actions.
+
+This and following sections provide more detailed information
+about common configuration options and conventions to implement the release
+process.
+
+All workflows require references to credentials for a remote registry.
+Typically, users configure them as [GitHub Actions Secrets], and the workflow
+performs authentication via the`docker/login-action`
+[action](http://github.com/docker/login-action).
+
+For example, adding the following step to a pipeline authenticates
+the job to `ghcr.io` using the workflow's ephemeral GitHub OIDC token.
+
+```yaml
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+```
+
+{{< hint "important" >}}
+By default, the job's OIDC token don't have permission to write packages
+to `ghcr.io`. Permissions are configurable in the GitHub repository's settings
+or declared
+[explicitly](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token)
+in the workflow definition YAML file.
+{{< /hint >}}
+
+For other registries, it's still best practice to reference credentials as
+custom Secret variables. For example:
+
+```yaml
+      - name: Login to Another Registry
+        uses: docker/login-action@v3
+        with:
+          registry: my-registry.io
+          username: ${{ env.REGISTRY_USER }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+```
+
+## Branching conventions
+
+Repositories for Crossplane extensions follow similar branching conventions
+to upstream Crossplane, where the release process assumes the workflow
+executing in branches with the `release-*` prefix. `main` is often included,
+though a conventional release process would not build and push off of tags on
+`main`.
+
+```yaml
+on:
+  push:
+    branches:
+      - main
+      - release-*
+```
+
+For example, when releasing `v0.1.0` of an extension, the conventional
+process is to cut a release branch `release-0.1` at the git commit
+where it builds from, and tag it as `v0.1.0`.
+
+{{< hint "note" >}}
+Some custom workflows may accept an explicit input for the remote reference,
+which overrides inferring from a git ref or tag. The [`ci.yml`](https://github.com/crossplane-contrib/function-python/blob/main/.github/workflows/ci.yml)
```

Comment:
depending on if the function workflows actually infer a version from a git tag/ref, consider updating this wording too